src/lib/hooks/useBlockNumber.tsx

```diff
@@ -63,6 +71,12 @@ export function BlockNumberProvider({ children }: { children: ReactNode }) {
           console.error(`Failed to get block number for chainId ${activeChainId}`, error)
         })
 
+      if (activeChainId === ChainId.MAINNET) {
+        setEthBlockNumber(block)
+      } else {
+        RPC_PROVIDERS[ChainId.MAINNET].getBlockNumber().then(setEthBlockNumber)
```

Comment:
I don't think you need to invoke this at all when `activeChainId !== ChainId.MAINNET`.
wdyt about just removing this line?

Reply:
i think the `else` is still required if we load initially onto another chain. I just tested it and it works fine if we connect to `mainnet` first and then switch chains, but if we for example connect to `Polygon` first then it doesn't load the ENS